### ISSUE TITLE
first() sends additional notifications if observer triggers new values

### ISSUE THREAD

**Issue by bman654** [Reactive-Extensions/RxJS#721]
This code causes an infinite loop:

```javascript
var s = new Rx.Subject();
s.first().subscribe(function (value) { s.onNext(value + 1); });
s.onNext(1);
```

See this jsFiddle: http://jsfiddle.net/bman654/4m17fke5/

**Comment by paulpdaniels** [comment]
See my answer to #720 and this [implementation](https://github.com/Reactive-Extensions/RxJS/blob/master/src/core/linq/observable/_firstordefault.js). This is the same recursion issue.

**Comment by bman654** [comment]
yeah I know it isn't a good pattern.  But it did have a surprising consequence when it occurred...

**Comment by mattpodwysocki** [comment]
@bman654 Fixed as per Issue #720
